src/server/services/recommendationService.ts

```diff
@@ -0,0 +1,97 @@
+import { getGroupPrefs, getPref } from './prefService';
+import PreferenceObject from '../classes/preferenceObject';
+import RecommendationObject from '../classes/recommendationObject';
+import {
+  algorithm, movieType, tvShowType, animeTVType, animeMovieType,
+} from './algorithmService';
+
+const compileGroupPreferences = async (id: string) => {
+  const groupPreferences = await getGroupPrefs(id);
+  const likedGenreCount = new Map();
+  const dislikedGenreCount = new Map();
+  let sumPrefRating = 0;
+  let sumMoviePref = 0;
+  let sumTVPref = 0;
+  let sumAnimePref = 0;
+  groupPreferences.forEach((preference) => {
+    // Entertainment type preferences
+    sumMoviePref += preference.moviePreference;
+    sumTVPref += preference.tvShowPreference;
+    sumAnimePref += preference.animePreference;
+
+    // Rating preference
+    sumPrefRating += preference.preferredRatings;
+
+    // Genre preferences
+    preference.likedGenres.forEach((genre) => {
+      if (likedGenreCount.has(genre)) {
+        likedGenreCount.set(genre, likedGenreCount.get(genre) + 1);
+      } else {
+        likedGenreCount.set(genre, 1);
+      }
+    });
+    preference.dislikedGenres.forEach((genre) => {
+      if (dislikedGenreCount.has(genre)) {
+        dislikedGenreCount.set(genre, dislikedGenreCount.get(genre) + 1);
+      } else {
+        dislikedGenreCount.set(genre, 1);
+      }
+    });
+  });
+  const finalLikedGenres: string[] = [];
+  const finalDislikedGenres: string[] = [];
+  likedGenreCount.forEach((likedGenre: string, LGCount: number) => {
```

Comment:
could be done more efficiently (in a future refactoring PR) to do this rather than nested foreach loops. Eg. concatenate both maps, for duplicate genres, we compute the different for the genre count. 

Since genre array sizes are generally small, won't much of an impact for now